### PR TITLE
Feature/adc/#71 create search route

### PR DESCRIPTION
**Pull request by AmauryDanisCousandier**
route /search créée avec plusieurs endpoints pour plus particulierement chercher des 'songs' mais aussi quelque routes pour acceder aux albums, artists et genres par mot clefs

### CODE REVIEW

File: .github/workflows/front-docker.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# on:
+#   push:
+#     branches:
+#       - 'main'
+#   pull_request:
+#     branches: [ main ]
+#   workflow_dispatch:
+
+# jobs:
+#   build_front:
```
Comment (Octopus773) [inline .github/workflows/front-docker.yml:12]: why do you add a CI file in this PR ?

File: front/Dockerfile.test [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# run tests
+FROM node:17
+WORKDIR /app
+COPY package.json yarn.lock ./
+RUN yarn install
+RUN yarn global add expo-cli
+RUN yarn test
```
Comment (Octopus773) [inline front/Dockerfile.test:7]: what is this ?

File: front/views/AuthenticationView.tsx [hunk 3]
```diff
@@ -32,6 +35,7 @@ const AuthenticationView = () => {
 	const [mode, setMode] = React.useState("signin" as "signin" | "signup");
 
 	return (
+		<Provider store={store}>
```
Comment (Octopus773) [inline front/views/AuthenticationView.tsx:38]: The provider is in App.tsx, you needed a workaround ?

File: front/views/__snapshots__/AuthenticationView.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`AuthenticationView Component view should render 1`] = `[Function]`;
```
Comment (Octopus773) [inline front/views/__snapshots__/AuthenticationView.test.tsx.snap:3]: What is this ?, the same question for all three `.snap` files

File: back/src/search/search.controller.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	@ApiOperation({
+		summary: 'Get songs details by advanced filter',
+		description: 'Get songs details by advanced filter',
+	})
+	@Post('song/advanced')
+	@HttpCode(200) // change from '201 created' to '200 OK' http default response code
+	async findAdvanced(@Body() searchSongDto: SearchSongDto): Promise<Song[] | null> {
+		try {
+			const ret = await this.searchService.findAdvanced({
+				albumId: searchSongDto.album ? + searchSongDto.album : undefined,
```
Comment (AnonymusRaccoon) [inline back/src/search/search.controller.ts:33]: This syntax is a bit hard to read, the +. Just run prettier (I think we should add this in the CI)

**Review by AnonymusRaccoon — COMMENTED** [review]
Why are all the tests files commented or tests weird things (like if the type of component is a function)? If you don't want to test anything, delete the files it would be easier to read,

File: front/views/AuthenticationView.tsx (hunk 3 above)
Comment (AmauryDanisCousandier) [reply front/views/AuthenticationView.tsx:38]: for previous unit tests strategy, removed it thanks

File: front/views/__snapshots__/AuthenticationView.test.tsx.snap (hunk 4 above)
Comment (AmauryDanisCousandier) [reply front/views/__snapshots__/AuthenticationView.test.tsx.snap:3]: same as the above comment, thought it was still useful but indeed it is not so removed it, thanks

**Comment by Octopus773** [comment]
@AmauryDanisCousandier Please delete the front/Dockerfile.test from this PR for the back

**Comment by AmauryDanisCousandier** [comment]
> Why are all the tests files commented or tests weird things (like if the type of component is a function)? If you don't want to test anything, delete the files it would be easier to read,

do you mean for front test files ?

**Comment by AmauryDanisCousandier** [comment]
> Why are all the tests files commented or tests weird things (like if the type of component is a function)? If you don't want to test anything, delete the files it would be easier to read,

@zoriya  do you mean for front test files ?

**Comment by zoriya** [comment]
> > Why are all the tests files commented or tests weird things (like if the type of component is a function)? If you don't want to test anything, delete the files it would be easier to read,
> 
> @zoriya  do you mean for front test files ?

I meant fronts files and robot files

**Review by Octopus773 — APPROVED** [review]

**Review by zoriya — APPROVED** [review]

**Pull request merged**